test/utils/image/manifest.go

```diff
@@ -148,3 +156,38 @@ func GetE2EImage(image Config) string {
 func GetPauseImageName() string {
 	return GetE2EImage(Pause)
 }
+
+// ReplaceRegistryInImageURL replaces the registry in the image URL with a custom one
+func ReplaceRegistryInImageURL(imageURL string) string {
+	parts := strings.Split(imageURL, "/")
+	countParts := len(parts)
+	registryAndUser := strings.Join(parts[:countParts-1], "/")
+
+	switch registryAndUser {
+	case "gcr.io/kubernetes-e2e-test-images":
+		registryAndUser = e2eRegistry
+	case "k8s.gcr.io":
+		registryAndUser = gcRegistry
+	case "gcr.io/k8s-authenticated-test":
+		registryAndUser = PrivateRegistry
+	case "gcr.io/google-samples":
+		registryAndUser = sampleRegistry
+	case "docker.io/library":
+		registryAndUser = dockerLibraryRegistry
+	case "gcr.io/gke-release":
+		registryAndUser = gcrReleaseRegistry
+	case "quay.io/k8scsi":
+		registryAndUser = quayK8sCSI
+	default:
+		if countParts == 1 {
+			// We assume we found an image from docker hub library
+			// e.g. openjdk -> docker.io/library/openjdk
+			registryAndUser = dockerLibraryRegistry
+			break
+		}
+
+		log.Printf("Unknown Registry: %s\n", registryAndUser)
```

Comment:
This will now show up in all logs (whether `KUBE_TEST_REPO_LIST` is used or not), right? That's probably okay, given that we want developers who run in a non-air-gapped cluster to write their test so that it can also run in an air-gapped cluster.

But given that, should this really just be a log message (which probably will get overlooked) and not an error?

Either way, it's not going to be obvious to developers who see this message a) where it comes from and b) what they need to do about it.

Can you add more details and instructions?

Reply:
Hm, I think I would favour a more descriptive error. I agree with you that probably nobody reads the log message if tests are passing.

Reply:
Sounds good to me.

Reply:
I updated the code, PTAL.